src/libraries/Common/src/System/Security/Cryptography/ECOpenSsl.cs

```diff
@@ -15,60 +15,40 @@ public ECOpenSsl(ECCurve curve)
             GenerateKey(curve);
         }
 
-        public ECOpenSsl(AsymmetricAlgorithm owner)
+        public ECOpenSsl(int keySizeBits)
         {
-            _key = new Lazy<SafeEcKeyHandle>(() => GenerateKeyLazy(owner));
+            _key = new Lazy<SafeEcKeyHandle>(() => GenerateKeyByKeySize(keySizeBits));
         }
 
         public ECOpenSsl(ECParameters ecParameters)
         {
             ImportParameters(ecParameters);
         }
 
+        // Takes ownership of the key
         public ECOpenSsl(SafeEcKeyHandle key)
         {
             _key = new Lazy<SafeEcKeyHandle>(key);
         }
 
         internal SafeEcKeyHandle Value => _key.Value;
 
-        private static SafeEcKeyHandle GenerateKeyLazy(AsymmetricAlgorithm owner) =>
-            GenerateKeyByKeySize(owner.KeySize);
-
         public void Dispose()
         {
             FreeKey();
         }
 
         internal int KeySize => Interop.Crypto.EcKeyGetSize(_key.Value);
 
-        internal SafeEvpPKeyHandle UpRefKeyHandle()
+        internal SafeEvpPKeyHandle CreateKeyHandle()
```

Comment:
I like that it's honest that it's creating a thing (looking at you, RSACng.Key), but it's a little... generic in the name.  Maybe this goes away in "round 3" (RSA was round 1), so maybe it's OK, but really it's about the fact that it's translating an EC_KEY into an EVP_PKEY, so "CreateEvpPKeyHandle" or "CreatePKeyHandle" or "WrapAsPKey" or something...

None of them feel solid, so if you're happy with the name I can't complain too much.

Reply:
I ended up with CreateEvpPKeyHandle